### ISSUE TITLE
Way to hide imports on "Find all references" ?

### ISSUE THREAD

**Issue by enomado** [rust-lang/rust-analyzer#13184]
Sorry for this question in issues, but may be there is a config option or so?

**Comment by lnicola** [comment]
No there isn't.

**Comment by enomado** [comment]
Looks like I made it.

src/references.rs/find_all_refs

```


match &mut yolo {
        Some(h) => {
            for l in h {
                for (file_id, rr) in &mut l.references {
                    let lema = sema.parse(file_id.clone()).syntax().clone();

                    rr.retain(|(rrr, cat)| {
                        let y = lema.child_or_token_at_range(rrr.clone()).unwrap();

                        let sdd = match y {
                            syntax::NodeOrToken::Node(x) => {
                                let p = x.kind();

                                if matches!(p, USE) {
                                    // dbg!(&x);
                                    dbg!("MATCH");
                                    false
                                } else {
                                    true
                                }

                                // dbg!(format!("{}", &x));
                            }
                            syntax::NodeOrToken::Token(x) => {
                                //
                                // x.kind()
                                dbg!(&x);
                                false
                            }
                        };
                        sdd
                    });


                }
            }
        }
        None => {}
    };
```

Dont mind on naming.

Is my idea on filtering like that okay? I wanna do PR

**Comment by Veykril** [comment]
There is no need to use the text range to look up with, the references contain the syntax node that matches so you just have to traverse the tree upwards from the node to check if its an ident of a use path.

**Comment by enomado** [comment]
@Veykril 

If they do, this could be much simpler. .

`pub references: NoHashHashMap<FileId, Vec<(TextRange, Option<ReferenceCategory>)>>,`

only FileId and Range.

**Comment by Veykril** [comment]
I don't follow what you mean here

**Comment by enomado** [comment]
@Veykril 
>references contain the syntax node
I mean no. It do not

**Comment by Veykril** [comment]
But they do? https://github.com/rust-lang/rust-analyzer/blob/master/crates/ide-db/src/search.rs#L55-L63 contains an `ast::NameLike` which is the node.

Oh I see, you are trying to filter out too late, you'll need to filter in the `make_searcher` closure.

**Comment by enomado** [comment]
@Veykril Yep I didnt show a whole code. My fault. You are right. Thank you for suggestion. That variety of types, transformation confused me

**Comment by jplatte** [comment]
Can this be closed now with #13186 and #13228 merged?
